### PR TITLE
Minijinja v2 migration

### PR DESCRIPTION
**Pull request by lquerel**
Content of this PR:
- Migration to MiniJinja 2
- 2 new filters: required and optional 
- Documentation update

### CODE REVIEW

File: crates/weaver_forge/src/extensions/otel.rs [hunk 1]
```diff
@@ -2,11 +2,40 @@
...
+        if required.as_str() == Some("required") {
+            rv.push(value);
+        }
+    }
+    Ok(rv)
+}
+
+/// Filters the input value to only include the non-required "object".
+/// A optional object is one that has a field named "requirement_level" which is not "required".
+pub(crate) fn optional(input: Value) -> Result<Vec<Value>, minijinja::Error> {
```
Comment (jsuereth) [inline crates/weaver_forge/src/extensions/otel.rs:27]: I'm not sure this is a great name as it conflicts with `opt-in`, and we've had confusion around that already.

Maybe just "not_required"

File: crates/weaver_forge/src/extensions/otel.rs [hunk 2]
```diff
@@ -2,11 +2,40 @@
...
 
+use crate::config::CaseConvention;
+
+/// Filters the input value to only include the required "object".
+/// A required object is one that has a field named "requirement_level" with the value "required".
+pub(crate) fn required(input: Value) -> Result<Vec<Value>, minijinja::Error> {
+    let mut rv = vec![];
+
+    for value in input.try_iter()? {
+        let required = value.get_attr("requirement_level")?;
```
[the remark this answers is not in the document]
Comment (lquerel) [reply crates/weaver_forge/src/extensions/otel.rs:17]: Updated the comment + added a description in the README.md

File: crates/weaver_forge/src/extensions/otel.rs (hunk 1 above)
Comment (lquerel) [reply crates/weaver_forge/src/extensions/otel.rs:27]: Fixed

**Comment by jsuereth** [comment]
I manually tested this on https://github.com/open-telemetry/semantic-conventions/pull/1000 and it appears to not be a breaking change for JINJA templates (outside of some additional whitespace showing up, but I call that par for the course in JINJA).

**Review by jsuereth — APPROVED** [review]

**Pull request merged**